### PR TITLE
Fixed issue where when BaseUri is not provided and Uri is provided, N…

### PR DESCRIPTION
**Pull request by sauravkumarsahoo**
…ullReferenceException occurs.
Added support for .net6.0.

### CODE REVIEW

**Comment by waxtell** [comment]
Hello!  Thanks for contributing!  Are you able to provide an example that demonstrates the issue?

**Comment by sauravkumarsahoo** [comment]
Hello, Great work btw, this was very helpful to me and my project.

Here's the piece of code that I'm utilizing:
```csharp
c.AddXAmazonApiGatewayOperationAnnotations
            (
                op =>
                {
                    op.WithIntegration
                    (
                        intOpt =>
                        {
                            intOpt.Type = IntegrationType.aws_proxy;
                            intOpt.Uri = LambdaInvokeUrl;
                            intOpt.HttpMethod = HttpMethod.Post.Method;
                        }
                    );
                }
            );
```

Here, the Uri is the absolute invoke Uri that I need to have the Lambda calls invoked. Note that it's not a so called Uri, it's something like `arn:aws:apigateway:...`.

Without the change I made, this is the exception I get:
```
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=Swashbuckle.AWSApiGateway.Annotations
  StackTrace:
   at Swashbuckle.AWSApiGateway.Annotations.XAmazonApiGatewayOperationFilter.<Apply>g__CreateDefaultIntegrationOptions|2_1(OperationFilterContext ctx, String baseUri, <>c__DisplayClass2_0& ) in D:\Code\Freelance\EthicallyPlus\BackEnd.Proxy\Swashbuckle.AWSApiGateway.Annotations\XAmazonApiGatewayOperationFilter.cs:line 67
```

What I did was add the clause to check whether an explicit Uri is provided or not, and based on that go for building the BaseUri. Do let me know if there's a better solution to this, you'd know your work best.

File: Swashbuckle.AWSApiGateway.Annotations/XAmazonApiGatewayOperationFilter.cs [hunk 1]
```diff
@@ -67,9 +67,10 @@ XAmazonApiGatewayIntegrationOptions CreateDefaultIntegrationOptions(OperationFil
                 return new XAmazonApiGatewayIntegrationOptions
                 {
                     HttpMethod = ctx.ApiDescription.HttpMethod,
-                    Uri = Uri.IsWellFormedUriString(baseUri, UriKind.RelativeOrAbsolute)
+                    Uri = _options.IntegrationOptions.Uri == null ? Uri.IsWellFormedUriString(baseUri, UriKind.RelativeOrAbsolute)
                             ? new Uri(new Uri(baseUri), ctx.ApiDescription.RelativePath).ToString()
-                            : $@"{baseUri}{(baseUri.EndsWith("/") ? string.Empty : "/")}{ctx.ApiDescription.RelativePath}",
+                            : $@"{baseUri}{(baseUri.EndsWith("/") ? string.Empty : "/")}{ctx.ApiDescription.RelativePath}"
+                                : null,
```
Comment (waxtell) [inline Swashbuckle.AWSApiGateway.Annotations/XAmazonApiGatewayOperationFilter.cs]: What do you think of this change?
```csharp
                    Uri = !string.IsNullOrWhiteSpace(_options.IntegrationOptions.Uri) 
                        ? _options.IntegrationOptions.Uri
                        : Uri.IsWellFormedUriString(baseUri, UriKind.RelativeOrAbsolute)
                            ? new Uri(new Uri(baseUri), ctx.ApiDescription.RelativePath).ToString()
                            : $@"{baseUri}{(baseUri?.EndsWith("/") ?? false ? string.Empty : "/")}{ctx.ApiDescription.RelativePath}",
```

I added the BaseUri property to allow for the automatic generation of a Uri if none were specified.  To that end, I like the idea of checking for the presence of a valid Uri first and then building one dynamically if necessary.  The null reference is avoided with the addition of the conditional null operator.

File: Swashbuckle.AWSApiGateway.Annotations/Swashbuckle.AWSApiGateway.Annotations.csproj [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;netstandard2.1;net5.0</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;netstandard2.1;net5.0;net6.0</TargetFrameworks>
```
Comment (waxtell) [inline Swashbuckle.AWSApiGateway.Annotations/Swashbuckle.AWSApiGateway.Annotations.csproj:3]: Please bump the version and notes 👍🏼

**Comment by sauravkumarsahoo** [comment]
Have updated as you shared, looks good. Also updated the dependencies (myself am using the latest version in my project and am not seeing any problems so far).

File: Swashbuckle.AWSApiGateway.Annotations/Swashbuckle.AWSApiGateway.Annotations.csproj [hunk 3]
```diff
@@ -26,13 +27,13 @@
...
-    <Version>2.4.1</Version>
+    <Version>2.5.0</Version>
     <PackageProjectUrl>https://github.com/waxtell/Swashbuckle.AWSApiGateway.Annotations</PackageProjectUrl>
     <RepositoryUrl>https://github.com/waxtell/Swashbuckle.AWSApiGateway.Annotations.git</RepositoryUrl>
     <RepositoryType>git</RepositoryType>
     <OpenApiGenerateDocuments>false</OpenApiGenerateDocuments>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Swashbuckle.AspNetCore" Version="5.0.0" />
+    <PackageReference Include="Swashbuckle.AspNetCore" Version="6.4.0" />
```
Comment (waxtell) [inline Swashbuckle.AWSApiGateway.Annotations/Swashbuckle.AWSApiGateway.Annotations.csproj:38]: Interesting...  I could not build the solution until I added
```csharp
<PackageReference Include="Microsoft.AspNetCore.Cors" Version="2.2.0" />
```

Comment (sauravkumarsahoo) [reply Swashbuckle.AWSApiGateway.Annotations/Swashbuckle.AWSApiGateway.Annotations.csproj:38]: Ah, strange. I think something to do with the latest Swashbuckle.AspNetCore?

Comment (waxtell) [reply Swashbuckle.AWSApiGateway.Annotations/Swashbuckle.AWSApiGateway.Annotations.csproj:38]: I suspect so...  does it build for you without it?

Comment (sauravkumarsahoo) [reply Swashbuckle.AWSApiGateway.Annotations/Swashbuckle.AWSApiGateway.Annotations.csproj:38]: No, it doesn't, just confirmed. I've pushed in a change with the dependency.

**Review by waxtell — APPROVED** [review]
Nice work!  Thank you again for your contribution!

**Comment by sauravkumarsahoo** [comment]
Thank you, glad to be able to contribute!

**Comment by sauravkumarsahoo** [comment]
Hey Will, could you please publish the latest package?

**Comment by waxtell** [comment]
Done!

**Pull request merged**